integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/CredentialTest.java

```diff
@@ -1,43 +1,133 @@
-                    return entity;
-                })
+    public void testCreateDeleteCreate(MockedTestContext testContext) {
+        String credentialName = getNameGenerator().getRandomNameForMock();
+        testContext
+                .given(CredentialTestDto.class)
+                .withName(credentialName)
+                .when(CredentialTestAction::create)
+                .when(CredentialTestAction::list)
+                .then(CredentialTestAssertion.listContains(credentialName, 1))
+                .when(CredentialTestAction::delete)
+                .when(CredentialTestAction::list)
+                .then(CredentialTestAssertion.listContains(credentialName, 0))
+                .when(CredentialTestAction::create)
+                .when(CredentialTestAction::list)
+                .then(CredentialTestAssertion.listContains(credentialName, 1))
                 .validate();
     }
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    public void testCreateTwice(MockedTestContext testContext) {
+        String credentialName = getNameGenerator().getRandomNameForMock();
+        testContext
+                .given(CredentialTestDto.class)
+                .withName(credentialName)
+                .when(CredentialTestAction::create)
+                .when(CredentialTestAction::list)
+                .then(CredentialTestAssertion.listContains(credentialName, 1))
+                .withName(credentialName)
+                .when(CredentialTestAction::create, key(BAD_REQUEST_KEY))
+                .expect(BadRequestException.class, key(BAD_REQUEST_KEY))
+                .validate();
+    }
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    public void testModifyCredential(MockedTestContext testContext) {
+        String credentialName = getNameGenerator().getRandomNameForMock();
+        String modifiedDescription = "modified";
+        testContext
+                .given(CredentialTestDto.class)
+                .withName(credentialName)
+                .when(CredentialTestAction::create)
+                .when(CredentialTestAction::list)
+                .then(CredentialTestAssertion.listContains(credentialName, 1))
+                .withName(credentialName)
+                .withDescription(modifiedDescription)
+                .when(CredentialTestAction::modify)
+                .when(CredentialTestAction::list)
+                .then(CredentialTestAssertion.validateModifcation(modifiedDescription))
+                .validate();
+    }
+
+    @Test(dataProvider = INVALID_ATTRIBUTE_PROVIDER)
+    public void testCreateCredentialWithInvalidAttribute(MockedTestContext testContext,
+        String credentialName, String expectedErrorMessage) {
+        testContext
+                .given(CredentialTestDto.class)
+                .withName(credentialName)
+                .when(CredentialTestAction::create, key(BAD_REQUEST_KEY))
+                .expect(BadRequestException.class, exceptionConsumer(e -> {
```

Comment:
please extract this RunningParameter to a separate method, to make the given `expect` more readable

Reply:
no